packages/web3-providers-base/types.ts

```diff
@@ -0,0 +1,5 @@
+export interface ProviderOptions {
+    providerString: string
+    protectProvider: boolean
+    supportsSubscriptions: boolean
```

Comment:
Since you set defaults in the constructor you should consider adding the optional type `?` to the types with defaults:
```
supportsSubscriptions?: boolean
```

Reply:
No longer relevant - `supportsSubscriptions` removed